### ISSUE TITLE
CI: angular/size test occasionally flakes (and no way to debug/rerun)

### ISSUE THREAD

**Issue by benlesh** [angular/angular#30341]
### Problem

While working on https://github.com/angular/angular/pull/30321, I ran into an issue where the angular size test failed inexplicably, saying ~19k had been added to the todo bundle. That particular PR was a change to tests only, so should have have done anything to size.

- There was no way to debug this
- There was no simple way to rerun the test

@kara and I both looked to see if we could find the culprit, but it seemed like a flake.

### Workaround

I simply did a `git commit --amend` and changed the comment a little, therefor changing the SHA so I could force-push nad it would run again. (It passed the second time).

**Comment by alxhub** [comment]
I just merged #30372 which had this problem. I downloaded the test bundles built from this PR as well as a PR which was "known good":

```
   14783 bad-pr/hello_world.js
    5235 bad-pr/hello_world.js.br
   68216 bad-pr/todo.js
   20296 bad-pr/todo.js.br
   14799 good-pr/hello_world.js
    5243 good-pr/hello_world.js.br
   67710 good-pr/todo.js
   20173 good-pr/todo.js.br
```

So no appreciable difference in bundle size. My suspicion is that the "golden" file in the test sometimes comes up empty or near-empty (maybe there's an error in fetching it?) and that results in the message, which always seems to indicate that the delta is almost the whole file size.
